collector/powersupplyclass.go

```diff
@@ -0,0 +1,183 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !nopowersupplyclass
+// +build linux
+
+package collector
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs/sysfs"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	powerSupplyClassIgnoredPowerSupplies = kingpin.Flag("collector.powersupply.ignored-supplies", "Regexp of power supplies to ignore for powersupplyclass collector.").Default("^$").String()
+)
+
+type powerSupplyClassCollector struct {
+	subsystem      string
+	ignoredPattern *regexp.Regexp
+	metricDescs    map[string]*prometheus.Desc
+}
+
+func init() {
+	registerCollector("powersupplyclass", defaultEnabled, NewPowerSupplyClassCollector)
+}
+
+func NewPowerSupplyClassCollector() (Collector, error) {
+	pattern := regexp.MustCompile(*powerSupplyClassIgnoredPowerSupplies)
+	return &powerSupplyClassCollector{
+		subsystem:      "power_supply",
+		ignoredPattern: pattern,
+		metricDescs:    map[string]*prometheus.Desc{},
+	}, nil
+}
+
+func (c *powerSupplyClassCollector) Update(ch chan<- prometheus.Metric) error {
+	powerSupplyClass, err := getPowerSupplyClassInfo(c.ignoredPattern)
+	if err != nil {
+		return fmt.Errorf("could not get power_supply class info: %s", err)
+	}
+	for _, powerSupply := range powerSupplyClass {
+
+		for name, value := range map[string]*int64{
+			"authentic":                   powerSupply.Authentic,
+			"calibrate":                   powerSupply.Calibrate,
+			"capacity":                    powerSupply.Capacity,
+			"capacity_alert_max":          powerSupply.CapacityAlertMax,
+			"capacity_alert_min":          powerSupply.CapacityAlertMin,
+			"charge_control_limit":        powerSupply.ChargeControlLimit,
+			"charge_control_limit_max":    powerSupply.ChargeControlLimitMax,
+			"charge_counter":              powerSupply.ChargeCounter,
+			"charge_empty":                powerSupply.ChargeEmpty,
+			"charge_empty_design":         powerSupply.ChargeEmptyDesign,
+			"charge_full":                 powerSupply.ChargeFull,
+			"charge_full_design":          powerSupply.ChargeFullDesign,
+			"charge_now":                  powerSupply.ChargeNow,
+			"charge_term_current":         powerSupply.ChargeTermCurrent,
+			"constant_charge_current":     powerSupply.ConstantChargeCurrent,
+			"constant_charge_current_max": powerSupply.ConstantChargeCurrentMax,
+			"constant_charge_voltage":     powerSupply.ConstantChargeVoltage,
+			"constant_charge_voltage_max": powerSupply.ConstantChargeVoltageMax,
+			"current_avg":                 powerSupply.CurrentAvg,
+			"current_boot":                powerSupply.CurrentBoot,
+			"current_max":                 powerSupply.CurrentMax,
+			"current_now":                 powerSupply.CurrentNow,
```

Comment:
We can drop `_now` from the name, and call it `current_ampers`, as this is the unit.

Reply:
Done that for every `*_now`